spinn_front_end_common/interface/abstract_spinnaker_base.py

```diff
@@ -249,6 +247,27 @@ def __signal_handler(self, _signal, _frame):
         logger.error("User has cancelled simulation")
         self._shutdown()
 
+    @property
+    def __bearer_token(self):
+        """
+        :return: The OIDC bearer token
+        :rtype: str or None
+        """
+        # Try using Jupyter if we have the right variables
+        jupyter_token = os.getenv("JUPYTERHUB_API_TOKEN")
+        jupyter_ip = os.getenv("JUPYTERHUB_SERVICE_HOST")
+        jupyter_port = os.getenv("JUPYTERHUB_SERVICE_PORT")
```

Comment:
Do we need to check all of these - surely one of them (the token?) will suffice?

Reply:
I copied this from the HBP code - this should work on the EBRAINS jupyter.  The "jupyter_token" is used as authorization against the jupyter port and ip to get the real access token (see the code below).